src/complex-list-field/complex-list-field.component.ts

```diff
@@ -151,23 +149,56 @@ export class ComplexListFieldComponent extends AbstractListFieldComponent implem
 
   onPageChange(page: number) {
     this.currentPage = page;
-    this.paginatedIndices = this.getIndicesForPage(page);
+    this.paginatedItems = this.getItemsForPage(page);
   }
 
-  getIndicesForPage(page: number): Array<number> {
-    let start = (page - 1) * this.navigator.itemsPerPage;
-    let indices: Array<number> = Array.apply(0, Array(this.navigator.itemsPerPage))
-      .map((el, index) => start + index);
-    // check if the indices includes some numbers that are out of values index range.
-    let lastIndexDiff = indices[indices.length - 1] - (this.values.size - 1);
-    if (lastIndexDiff > 0) {
-      indices.splice(indices.length - lastIndexDiff);
-    }
-    return indices;
-  }
+  getItemsForPage(page: number): Array<PaginatedItem> {
+    let items: Array<PaginatedItem> = [];
+    let indices: Array<number>;
+    indices = this._getIndicesToDisplay(page);
 
+    items = indices.map((el) => {
+      let isEditFormVisible = false;
+      let showEditForm = this.schema['showEditForm'];
+      if (showEditForm === undefined || showEditForm(this.values.get(el))) {
+        isEditFormVisible = true;
+      }
+      return {
```

Comment:
if you change `el` to `index` then you could use object literal shorthand here `{ index, isEditFormVisible }`